shared/mad/codeql/mad/modelgenerator/internal/ModelGeneratorImpl.qll

```diff
@@ -658,20 +657,40 @@ module MakeModelGenerator<
     ) {
       PropagateContentFlow::flow(p, reads, returnNodeExt, stores, preservesValue) and
       returnNodeExt.getEnclosingCallable() = api and
-      p.(NodeExtended).getEnclosingCallable() = api
+      p.(NodeExtended).getEnclosingCallable() = api and
+      p = api.getARelevantParameterNode()
     }
 
     /**
      * Holds if any of the content sets in `path` translates into a synthetic field.
      */
     private predicate hasSyntheticContent(PropagateContentFlow::AccessPath path) {
-      exists(PropagateContentFlow::AccessPath tail, DataFlow::ContentSet head |
-        head = path.getHead() and
-        tail = path.getTail()
-      |
-        exists(getSyntheticName(head)) or
-        hasSyntheticContent(tail)
-      )
+      exists(getSyntheticName(path.getAtIndex(_)))
+    }
+
+    private string getHashAtIndex(PropagateContentFlow::AccessPath ap, int i) {
+      result = getSyntheticName(ap.getAtIndex(i))
+    }
+
+    private string getReversedHash(PropagateContentFlow::AccessPath ap) {
+      result = concat(int i | | getHashAtIndex(ap, i), "" order by i desc)
```

Comment:
Use an actual separator instead of `""`?

Reply:
Sure, I will add a `.`, but we never really print them.